common/blockdevice/util.go

```diff
@@ -256,6 +256,18 @@ func GetDriveType(obj unstructured.Unstructured) (string, error) {
 	return unstruct.GetString(&obj, "spec", "details", "driveType")
 }
 
+// IsReserved checks if block device is resrved or not
+// using openebs.io/block-device-tag label
+func IsReserved(obj unstructured.Unstructured) (bool, error) {
+	if obj.GetKind() != string(types.KindBlockDevice) {
+		return false,
+			errors.Errorf("Can not check block device is reserved or not: Expected kind %q got %q",
+				types.KindBlockDevice, obj.GetKind())
+	}
+	_, ok := obj.GetLabels()["openebs.io/block-device-tag"]
```

Comment:
nil pointer check?

Reply:
added test case

Reply:
`GetLabels()` will not return nil map it returns empty map is key not found.